jl4-lsp/src/LSP/Core/Types/Options.hs

```diff
@@ -117,7 +117,14 @@ defaultIdeOptions = IdeOptions
     ,optSkipProgress = defaultSkipProgress
     ,optMaxDirtyAge = 100
     ,optRunSubset = True
-    ,optPreservedKeys = \_ -> PreservedKeys HashSet.empty
+    ,optPreservedKeys = \_ -> PreservedKeys $ HashSet.fromList
+        -- always preserved
+        [ typeOf GetFileExists
+        , typeOf GetModificationTime
+        , typeOf IsFileOfInterest
+        , typeOf GetClientSettings
+        , typeOf AddWatchedFile
+        ]
```

Comment:
what does this do?

Reply:
Avoid garbage collection of important keys

This avoids overly-eager garbage collection of results for certain keys.
All the keys we preserve are taken from HLS.

This is mostly me sneakily adding orthogonal commits to PRs out of laziness.

Reply:
No this is fine, I just wanted to know what it does. Thanks!